### DISCUSSION TITLE
multipart List[float] 422 bugged?

### DISCUSSION THREAD

**Discussion by tamis-laan** [fastapi/fastapi#8604] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python javascript
@router.post('/add',status_code=200)
async def add(values: List[float] = Form(...), file: UploadFile = File(...)):
    ...
```

### Description

Given a post route with a file and array of floats when using the swagger interface '/docs' and posting a file and an array with a SINGLE item everything works. Posting an file and an array with MORE then one value fastAPI returns `422 Unprocessable Entity`.

Working CURL request:
``` console
curl -X 'POST' \
  'http://localhost:8000/add' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'file=@image.jpg;type=image/jpeg' \
  -F 'value=1'  // <-------------------
```

Failing CURL request:
``` console
curl -X 'POST' \
  'http://localhost:8000/add' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'file=@image.jpg;type=image/jpeg' \
  -F 'value=1,2' // <-------------------
```

### Operating System

Linux

### FastAPI Version

0.74.1

### Python Version

3.9.10

**Comment by tamis-laan** [comment]
If I change `List[float] = Form(...)` to `List = Form(...)` and use the swagger interface with: 
``` console
curl -X 'POST' \
  'http://localhost:8000/add' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'values=1,2,3' \
  -F 'file=@file.jpg;type=image/jpeg'
```
I get the following:
``` javascript
> print(values)
['1,2,3']
```
Clearly wrong...

How do I send a file and an array using fastAPI?

**Comment by gmanny** [comment, marked as answer]
What you can do is send a list of parameters the same way as [a list of Query parameters](https://fastapi.tiangolo.com/tutorial/query-params-str-validations/?h=list#query-parameter-list-multiple-values). Like this:

```curl
curl -X POST http://localhost:8000/add -H "accept: application/json" -H "Content-Type: multipart/form-data" -F "values=1" -F "values=2" -F "file=@file.png;type=image/png"
```

The html form for this would look something like this:

```html
<form enctype="multipart/form-data" action="http://localhost:8000/add" method="post"><input type="submit">Send</input><input type="hidden" name="values" value="1.2" /><input type="hidden" name="values" value="2.2" /><input type="file" name="file" /></form>
```

**Comment by gmanny** [comment]
It's overall a convention to treat form parameters the same as query parameters. I know PHP kind of does the same.
